src/Chapter3.hs

```diff
@@ -971,6 +1073,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Show, Enum, Eq, Bounded)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend day = case day of
+  Saturday -> True
+  Sunday   -> True
+  _        -> False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+  | day == maxBound = minBound
```

Comment:
Great! 🎉 
Using `Bounded` in here makes code more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂